crates/sui/tests/reconfiguration_tests.rs

```diff
@@ -353,23 +353,30 @@ async fn test_reconfig_with_committee_change_basic() {
         new_validator.protocol_key.concise()
     );
 
-    let sender = new_node_config.sui_address();
-    let gas = Object::with_owner_for_testing(sender);
-    let stake = Object::with_owner_for_testing(sender);
+    let new_validator_address = new_node_config.sui_address();
+    let genesis_gas = Object::with_owner_for_testing(new_validator_address);
+    let stake = Object::new_gas_with_balance_and_owner_for_testing(
+        25_000_000_000_000_000,
+        new_validator_address,
+    );
 
-    let mut authorities =
-        spawn_test_authorities([gas.clone(), stake.clone()].into_iter(), &init_configs).await;
+    let mut authorities = spawn_test_authorities(
+        [genesis_gas.clone(), stake.clone()].into_iter(),
+        &init_configs,
+    )
+    .await;
 
     let proof_of_possession =
-        generate_proof_of_possession(new_node_config.protocol_key_pair(), sender);
+        generate_proof_of_possession(new_node_config.protocol_key_pair(), new_validator_address);
 
-    let tx_data = TransactionData::new_move_call_with_dummy_gas_price(
-        sender,
+    // Step 1: Add the new node as a validator candidate.
```

Comment:
@lxfind Do you see this validator joining flow being used in other future tests too? If so I'll put the three steps in a helper fun